src/Nethermind/Nethermind.TxPool/Collections/TxDistinctSortedPool.cs

```diff
@@ -58,7 +58,7 @@ protected override void UpdateGroup(Address groupKey, EnhancedSortedSet<Transact
                 {
                     bool reAdd = _worstSortedValues.Remove(tx);
                     change(tx);
-                    if (reAdd)
+                    if (reAdd && tx.Hash != null)
```

Comment:
do we expect null tx.Hash here? Is this correct behavior? shouldn't `tx.Hash!` be used here?